book/src/proptest-derive/modifiers.md

```diff
@@ -0,0 +1,370 @@
+
+Each modifier within the parentheses is independent, in that putting two
+modifiers in the same attribute is equivalent to having two `#[proptest(..)]`
+attributes with one modifier each.
+
+For brevity, modifiers are sometimes referenced by name alone; e.g., "the
+`weight` modifier" refers to `#[proptest(weight = nn)]` and not some
+freestanding `#[weight]` attribute.
+
+## `filter`
+
+Form: `#[proptest(filter = F)]` or `#[proptest(filter(F))]` where `F` is either
+a bare identifier or a Rust expression in a string. In either case, the
+parameter must evaluate to something which is `Fn (&T) -> bool`, where `T` is
+the type of the item being filtered.
+
+Usable on: structs, enums, enum variants, fields
+
+The `filter` modifier allows filtering values generated for a field via
+rejection sampling. Since rejection sampling is inefficient and interferes with
+shrinking, it should only be used for conditions that are very rare or are
+unfeasible to express otherwise. In many cases, [`strategy`](#strategy) can be
+used to more directly express the desired behaviour without rejection sampling.
+See the documentation for [`prop_filter`] for more details.
+
+The argument to the modifier must be a valid argument for the second parameter
+of [`prop_filter`].
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(filter = "|segment| segment.start != segment.end")]
+struct NonEmptySegment {
+    start: i32,
+    end: i32,
+}
+```
+
+As mentioned above, filtering should be avoided when it is reasonably possible
+to express a non-filtering strategy that achieves the same effect. For example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct BadExapmle {
+    // Don't do this! Your tests will run more slowly and shrinking won't work
+    // properly.
+    #[proptest(filter = "|x| x % 2 == 0")]
+    even_number: u32,
+}
+
+#[derive(Debug, Arbitrary)]
+struct GoodExample {
+    // Directly generate even numbers only by transforming the set of all
+    // `u32`s and then mapping it to the set of even `u32`s.
+    #[proptest(strategy = "any::<u32>().prop_map(|x| x / 2 * 2)")]
+    even_number: u32,
+}
+```
+
+[`prop_filter`]: https://altsysrq.github.io/rustdoc/proptest/latest/proptest/strategy/trait.Strategy.html#method.prop_filter
+
+## `no_bound`
+
+Form: `#[proptest(no_bound)]`
+
+Usable on: generic type declarations
+
+Normally, when `#[derive(Arbitrary)]` is applied to an item with generic type
+arguments, every type argument is required to `impl Arbitrary`. For example,
+given a declaration like the following:
+
+```rust
+#[derive(Arbitrary)]
+struct MyStruct<T> { /* ... */ }
+```
+
+Something like this will be generated:
+
+```rust
+impl<T> Arbitrary for MyStruct<T> where T: Arbitrary { /* ... */ }
+```
+
+Placing `#[proptest(no_bound)]` on a generic type argument suppresses this. For
+example, the following removes the extra `where T: Arbitrary`:
+
+```rust
+#[derive(Arbitrary)]
+struct MyStruct<#[proptest(no_bound)] T> { /* ... */ }
+```
```

Comment:
I'd also include an example:


```rust
#[derive(Arbitrary)]
#[proptest(no_bound)]
struct MyStruct<T, U> { /* ... */ }
```

since that short-hand also exists.